### PR TITLE
[SIO-220] Scrape places rating to a file periodically

### PR DESCRIPTION
**Pull request by notgne2**
Not yet tested, rough idea. Putting the place ID in the Nix feels dirty and wrong, no idea if the systemd logic there is sane or not

### CODE REVIEW

File: lib/modules/website.nix [hunk 1]
```diff
@@ -170,7 +171,31 @@ in {
         --scripts-prepend-node-path=true \
         --serokell-website:port=${toString port} \
         --serokell-website:uploadpath=/var/lib/${name}/files
+        --serokell-website:placesrating=/var/lib/${name}/places-rating.json
     '';
   };
 
+  systemd.services."${name}-places-rating" = {
+    requires = [ "network-online.target" "${name}" ];
+    after = [ "network-online.target" "${name}" ];
```
Comment (notgne2) [inline lib/modules/website.nix:180]: I guess `network-online.target` here is redundant?

File: lib/modules/website.nix [hunk 2]
```diff
@@ -170,7 +171,31 @@ in {
         --scripts-prepend-node-path=true \
         --serokell-website:port=${toString port} \
         --serokell-website:uploadpath=/var/lib/${name}/files
+        --serokell-website:placesrating=/var/lib/${name}/places-rating.json
     '';
   };
 
+  systemd.services."${name}-places-rating" = {
+    requires = [ "network-online.target" "${name}" ];
```
Comment (kirelagin) [inline lib/modules/website.nix:179]: This does not look right, I think it should be the opposite: the website unit pulls this one.

File: lib/modules/website.nix (hunk 1 above)
Comment (kirelagin) [reply lib/modules/website.nix:180]: I don’t really think we need any after/before dependency between this and www.

**Review by kirelagin — COMMENTED** [review]
I am not convinced that this amount of systemd logic is enough. First of all, we want to make sure that when the website is running, the rating will get updated (so, www should pull the timer, I suppose?). Also, regardless of the timer, we want to make sure that when the website starts, it has this file, so, I suppose, www should pull the oneshot service, at least if the file does not exist.

Ideally, the logic should be “if the file does not exist or does exist but is old, go create it; if it exists and is fresh enough, do nothing”. In practice, I think, even if the oneshot fires everytime www is started, it is fine? The point is that making this API query costs 2 cents or something like that, so I want some sort of guard that will make sure it does not get called hundreds of times a day.

File: lib/modules/website.nix [hunk 3]
```diff
@@ -170,7 +171,31 @@ in {
...
+      User = name;
+    };
+
+    script = ''
+      ${pkgs.curl}/bin/curl "https://maps.googleapis.com/maps/api/place/details/json?placeid=${placeid}&fields=rating,user_ratings_total&key=$(cat ${vs.${name}}/maps_api_key)" | ${pkgs.jq}/bin/jq .result > /var/lib/${name}/places-rating.json
+    '';
+  };
+
+  systemd.timers."${name}-places-rating" = {
+    wantedBy = [ "timers.target" ];
```
Comment (kirelagin) [inline lib/modules/website.nix:193]: I don’t think we need this? We want the service to be logically part of the website, and this, IIUC, just says that it should be running all the time, which, um, works I guess, but is not quite right logically.

**Comment by kirelagin** [comment]
Hello?

**Comment by notgne2** [comment]
@kirelagin I've switched it around so that the website now depends on it instead, and adapted the script to check if the file is missing or too old before charging us money

**Comment by notgne2** [comment]
Making this not a draft given that you didn't seem to have an issue with the file placement

**Comment by rvem** [comment]
The corresponding issue was aborted, so I'm closing this PR now

**Pull request closed without merging**